File: projects/10/ruby/Kiwi.rb
#!/usr/bin/ruby
# Kiwi.rb Defines the Kiwi class which inherits attributes and methods from the Bird superclass.
# Begun by: Dr. Adams, for CS 214 at Calvin College
# Completed by: Ian Christensen, for CS 214 at Calvin College
# Date: Spring, 2018

require './Walking.rb'

class Kiwi < Walking

	# Method: call, accessor for a Duck call
	# Precondition: Duck must exist
	# Receives: a Duck object
	# Returns: 'Quack!', a duck call
	def call
		'Aaawhhh!!!'
	end
end

File: projects/10/java/Kiwi.java
/* Kiwi.java provides a Kiwi subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Ian Christensen, for CS 214 at Calvin College
 * Date: Spring, 2018
 */

public class Kiwi extends Walking {

	public Kiwi() {
		super();
	}
    
	/* explicit constructor
	 * Receive: name, a String 
	 * PostCond: myName == name.
	 */
	public Kiwi(String name) {     
		super(name);
	} 

	/* A Kiwi's Call
	 * Return: a Kiwi-call ("Aaawhhh!!!").
	 */
	public String call() {
		return "Aaawhhh!!!";
	} 
}
